src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java

```diff
@@ -189,6 +214,39 @@ else if (newLine == null || tag.getLineNumber() - newLine.getLineNumber() != 1)
         }
     }
 
+    /**
+     * Determines whether or not the paragraph tag is followed by block tag.
+     *
+     * @param tag html tag.
+     * @return block tag if the paragraph tag is followed by block tag.
+     */
+    private static String checkIfFollowedByBlockTag(DetailNode tag) {
+        String blockTag = null;
+        DetailNode htmlElement = JavadocUtil.getNextSibling(tag);
+
+        while (htmlElement != null && htmlElement.getType() != JavadocTokenTypes.HTML_ELEMENT) {
+            if (htmlElement.getType() == JavadocTokenTypes.TEXT
+                    && !CommonUtil.isBlank(htmlElement.getText())) {
+                htmlElement = null;
+                break;
+            }
+            htmlElement = JavadocUtil.getNextSibling(htmlElement);
+        }
+
+        if (htmlElement != null) {
+            final DetailNode htmlTag = JavadocUtil.getFirstChild(htmlElement);
+            final DetailNode htmlTagFirstChild = JavadocUtil.getFirstChild(htmlTag);
+            for (DetailNode child: htmlTagFirstChild.getChildren()) {
+                final String htmlTagName = child.getText();
```

Comment:
move this to function getHtmlElementName
and you do not need FOR.

```
|--LEADING_ASTERISK ->  * [4:0]
|--TEXT ->   [4:2]
|--HTML_ELEMENT -> HTML_ELEMENT [4:3]
|   `--P_TAG_START -> P_TAG_START [4:3]
|       |--START -> < [4:3]
|       |--P_HTML_TAG_NAME -> p [4:4]
|       `--END -> > [4:5]
|--NEWLINE -> \n [4:6]
|--LEADING_ASTERISK ->  * [5:0]
|--TEXT ->   [5:2]
|--HTML_ELEMENT -> HTML_ELEMENT [5:3]
|   `--HTML_TAG -> HTML_TAG [5:3]
|       |--HTML_ELEMENT_START -> HTML_ELEMENT_START [5:3]
|       |   |--START -> < [5:3]
|       |   |--HTML_TAG_NAME -> table [5:4]
|       |   `--END -> > [5:9]

```

you can reuse https://github.com/checkstyle/checkstyle/blob/37a777c3f69b162e88bb6fc88659fe4595248b29/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtil.java#L198C30-L198C74